### ISSUE TITLE
pylint checker: add max_issues parameter for detailed + summary output

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#32]
## Summary

Enhance the pylint checker to show **full details for the most important issues** plus a **summary of all remaining issues**, instead of only reporting the first issue type found.

## Current Behavior

- `run_pylint_check` only returns details for the **first** issue type found (arbitrary order from Python `set`)
- No statistics or overview of all issue types
- No way to control how many issues get detailed output

### API Change

Add a single new parameter `max_issues` to `run_pylint_check`:

```python
run_pylint_check(
    extra_args: list[str] | None = None,        # unchanged
    target_directories: list[str] | None = None, # unchanged
    max_issues: int = 1,                          # NEW
)
```

### Behavior

- **`max_issues`** controls how many issue **types** get full details (fix instructions + locations)
- Issues are sorted by **severity** (fatal > error > warning > refactor > convention), then by **frequency** (most occurrences first)
- Top N issue types shown with full details + fix instructions
- Remaining issue types shown as a flat summary list, same sort order
- **Location cap**: Each detailed issue type shows at most **50 locations**. If more exist, append `"... and X more occurrences"`
- `max_issues=0` → statistics only: total count line + breakdown list of all issue types with occurrence counts (no fix instructions or locations)
- **Contextual hints in return values**: When useful, include actionable hints (e.g., `"Use max_issues=5 to see more details"`). No dedicated help parameter.

### Docstring

Slim down the `run_pylint_check` docstring to reduce always-sent tokens. Keep only one-liner description + minimal parameter descriptions. Examples and detailed explanations removed. The JSON schema already provides type/default info.

```python
"""
Run pylint on the project code and generate smart prompts for LLMs.

Args:
    extra_args: Additional pylint arguments.
    target_directories: Directories to analyze relative to project_dir. Defaults to ["src"] and "tests" if it exists.
    max_issues: Number of issue types to show in detail (default: 1). Remaining issues shown as summary counts.
"""
```

### Output Format Examples

**Default `max_issues=1`:**
```
pylint found some issues related to code E0602.
Ensure a variable is assigned before it is used...
[full details for E0602 locations, max 50]

--- 3 additional issue types found (14 occurrences) ---
- W0613 unused-argument: 4 occurrences
- C0411 wrong-import-order: 3 occurrences
- C0301 line-too-long: 7 occurrences
```

**`max_issues=0` (statistics only):**
```
pylint found 4 issue types (14 total occurrences):
- E0602 undefined-variable: 3 occurrences
- W0613 unused-argument: 4 occurrences
- C0411 wrong-import-order: 3 occurrences
- C0301 line-too-long: 7 occurrences
```

**Zero issues:**
```
Pylint passed. 0 issues in X files.
```
(If file count is not easily available from pylint, fall back to: `Pylint passed. 0 issues found.`)

## Constraints

- **No LLM-coaching text** in output (no "use report_mode=X for more details") — but contextual hints in return values are OK when directly relevant
- **Backward compatible** default — `max_issues=1` gives enhanced version of current behavior
- **File/folder filtering** already supported via `target_directories` parameter — no changes needed
- **Location cap**: 50 per issue type in detailed sections

## Files to Change

- `models.py` — helper methods for grouping/sorting by severity + frequency (severity priority mapping, sorted issue summary)
- `reporting.py` — new summary generation, modify `get_pylint_prompt()`, location cap logic
- `server.py` — add `max_issues` parameter to tool registration, slim down docstring
- Tests for all of the above

---

# Original issue: pylint checker - support for all warnings, or just next

How does the pylint checker work at the moment?
Does it just show one error type?
Or all found?
COuld there be a parameter to get all found?
Eg to understand which warnings are in the code base?

What does it do when there are many (eg more than 50) warnings?
Give a statistic by file and error type?

Is it possible to filter by one or several files / folders with subelements?

**Comment by MarcusJellinghaus** [comment]
Claude does summary statistics like this
```
cd /c/Users/Marcus/Documents/GitHub/mcp_coder && python -m pylint src/ --disable=all --enable=W --output-format=text 2>&1 | 
      grep "^src" | grep -oP 'W[0-9]+' | sort | uniq -c | sort -rn)
  ⎿      441 W1203
         176 W0718
          36 W0611
          11 W0212
           8 W0613
           7 W0707
           7 W0603
           4 W0612
           2 W4903
           2 W0706
           1 W1404
           1 W0719
```
